compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp

```diff
@@ -388,21 +388,23 @@ LogicalResult setMatmulOpConfig(linalg::LinalgOp op, int64_t subgroupSize,
 
   // Deduce the configuration for the K dimension. We need some power of two
   // here so that we can do vector load.
-  for (int64_t t = llvm::PowerOf2Floor(residualTilingFactor); t >= 2; t >>= 1) {
+  for (int64_t t = llvm::PowerOf2Floor(residualTilingFactor); t >= 1; t >>= 1) {
     if (dimK % t == 0) {
       reductionTileSizes[kIndex] = t;
       break;
     }
   }
   if (reductionTileSizes[kIndex] == 0) return success();
 
-  auto totalThreads =
-      std::accumulate(workgroupSize.begin(), workgroupSize.end(), 1,
-                      std::multiplies<int64_t>());
+  int64_t totalThreads = workgroupSize[0] * workgroupSize[1] * workgroupSize[2];
   LLVM_DEBUG({
     llvm::dbgs() << "total thread count = " << totalThreads << "\n";
     llvm::dbgs() << "subgroup size = " << subgroupSize << "\n";
   });
+  // If we cannot keep all the threads in a single subgroup activel, don't go
+  // through this pipeline.
+  if (totalThreads < subgroupSize) return success();
```

Comment:
Isn't this line handled by the the conditional operator of pipeline?
Also isn't subgroup size warp size? shouldn't just be the block/group size that matters for shared/workgroupMemory?